app/Http/Controllers/ScheduledMessageController.php

```diff
@@ -46,22 +49,26 @@ public function create(Request $request)
                 ->withInput();
         }
 
-        ScheduledMessage::create([
+        $message = ScheduledMessage::create([
             'body_en' => $request->input('body_en'),
             'body_es' => $request->input('body_es'),
             'target_sms' => $request->input('target_sms'),
             'target_twitter' => $request->input('target_twitter'),
             'send_at' => new Carbon($request->input('send_at')),
         ]);
 
+        $message->tags()->attach(array_keys($request->input('tags')));
```

Comment:
It looks like the `array_keys` here is causing your tests to fail. Removing `array_keys` makes the test green, I think that should work in the real app too right?

```suggestion
        $message->tags()->attach($request->input('tags'));
```